src/pg_to_evalscript/javascript_processes/add_dimension.js

```diff
@@ -1,29 +1,35 @@
 function add_dimension(arguments) {
   const { data, name, label, type = "other" } = arguments;
 
-  if (data === undefined) {
-    throw new Error("Mandatory argument `data` is not defined.");
-  }
-
-  if (name === undefined) {
-    throw new Error("Mandatory argument `name` is not defined.");
-  }
-
-  if (label === undefined) {
-    throw new Error("Mandatory argument `label` is not defined.");
-  }
-
-  if (typeof name !== "string") {
-    throw new Error("Argument `name` is not a string.");
-  }
-
-  if (typeof label !== "number" && typeof label !== "string") {
-    throw new Error("Argument `label` is not a string or a number.");
-  }
-
-  if (typeof type !== "string") {
-    throw new Error("Argument `type` is not a string.");
-  }
+  validateParameter({
+    processName: "add_dimension",
+    parameterName: "data",
+    value: data,
+    required: true,
+  });
+
+  validateParameter({
+    processName: "add_dimension",
+    parameterName: "name",
+    value: name,
+    required: true,
+    allowedTypes: ["string"],
+  });
+
+  validateParameter({
+    processName: "add_dimension",
+    parameterName: "label",
+    value: label,
+    required: true,
+    allowedTypes: ["string", "number"],
+  });
+
+  validateParameter({
+    processName: "add_dimension",
+    parameterName: "type",
+    value: type,
+    allowedTypes: ["string"],
```

Comment:
Possible future improvement could be to add the allowed values

e.g. the `type` has these allowed values: `spatial`, `temporal`, `bands`, other`

It's easy for string values and booleans, but might be harder for numbers (greater than, etc.)

FFTR.